### PR TITLE
Store and Sampling working together

### PR DESCRIPTION
**Pull request by aarati-K**
Work In Progress. Do not merge yet.

### CODE REVIEW

File: store/cifar10.py [hunk 1]
```diff
@@ -1,39 +1,125 @@
...
+        train_metadata = self.metadata.get(self.TRAIN_FOLDER)
+        if train_metadata:
+            self.key_size = train_metadata.get(MetadataField.KEY_SIZE)
+            self.value_size = train_metadata.get(MetadataField.VALUE_SIZE)
+        else:
+            self.key_size = self.value_size = None
 
     def count_num_points(self):
-        self.num_points = 50000
+        self.num_train_points = 50000
```
Comment (abhay-venkatesh) [inline store/cifar10.py:23]: I am assuming you are temporarily hardcoding this? Might want to do add a TODO: to make that clear

File: store/cifar10.py [hunk 2]
```diff
@@ -1,39 +1,125 @@
...
-                        fo = open(path, 'rb')
-                        p = pickle.load(fo, encoding='bytes')
-                        data = np.array(p.get(b'data'))
-                        labels = np.array(p.get(b'labels'))
-                        data_points= []
-                        for data_point, label in zip(data,labels):
-                            data_points.append([data_point, label])
-                        nparr = np.array(data_points)
-                        np.save(f,nparr)
+        data_folder_path = self.get_data_folder_path()
```
Comment (abhay-venkatesh) [inline store/cifar10.py:27]: might want to move this into the __init__()

File: store/cifar10.py [hunk 3]
```diff
@@ -1,39 +1,125 @@
...
+                os.remove(train_file_path)
+            else:
+                msg += "Skipping."
+                print(msg)
+                return
+
+        # Generate train dataset
+        f_train = Path(train_file_path).open('ab')
+        for root, sub_dirs, files in os.walk(self.input_data_folder):
+            for file in files:
```
Comment (abhay-venkatesh) [inline store/cifar10.py]: might want to use a different name for file, its a builtin

**Comment by abhay-venkatesh** [comment]
some minor changes but looks fine overall

File: store/cifar10.py (hunk 1 above)
Comment (aarati-K) [reply store/cifar10.py:23]: For Cifar10 we already know the dataset size.. so didn't see a point in writing code.

File: store/cifar10.py (hunk 2 above)
Comment (aarati-K) [reply store/cifar10.py:27]: You mean the folder creation logic?

File: store/cifar10.py (hunk 3 above)
Comment (aarati-K) [reply store/cifar10.py]: Will do.

File: store/cifar10.py (hunk 2 above)
Comment (abhay-venkatesh) [reply store/cifar10.py:27]: or sure yea that works too

File: store/cifar10.py (hunk 1 above)
Comment (abhay-venkatesh) [reply store/cifar10.py:23]: oh yeah nvm

File: store/cifar10.py (hunk 2 above)
Comment (SrujithPoondla) [reply store/cifar10.py:27]: He is asking to initialize the data_folder_path in init i guess

File: store/cifar10.py [hunk 4]
```diff
@@ -1,39 +1,125 @@
...
+        test_file_path = test_folder_path + '/' + self.DATA_FILE.format(0)
+        f_test = Path(test_file_path).open('ab')
+        test_data_file = self.input_data_folder + "/test_batch"
+        nparr = self.read_cifar_batch(test_data_file)
+        np.save(f_test, nparr)
+
+        # Create metadata for train and test folders
+        self.write_metadata()
+
+    def read_cifar_batch(self, batch_filename):
```
Comment (SrujithPoondla) [inline store/cifar10.py]: Can you change this function name to parse_cifar_data? It is confusing whether we are reading from the dataset directory or getting the batch from shared memory?

Comment (aarati-K) [reply store/cifar10.py]: Will do.

Comment (SrujithPoondla) [reply store/cifar10.py]: You are saving each batch file separately as .npy file? i.e for this cifar-10 there will be 5 .npy files?

File: store/cifar10.py [hunk 5]
```diff
@@ -1,39 +1,125 @@
...
+
+        # Assign key and value size, if not already assigned
+        if not self.key_size:
+            self.key_size = 4 # bytes
+            # value size + 1 (for label)
+            self.value_size = (len(data_points[0][0]) + 1)*4 #bytes
+
+        return np.array(data_points)
+
+    def write_metadata(self):
```
Comment (aarati-K) [inline store/cifar10.py:90]: Move key size and value size fields to base class.

File: store/cifar10.py (hunk 4 above)
Comment (aarati-K) [reply store/cifar10.py]: Does `read_cifar_file` seem okay?

**Review by SrujithPoondla — COMMENTED** [review]
Looks good

File: store/cifar10.py (hunk 4 above)
Comment (SrujithPoondla) [reply store/cifar10.py]: yeah

File: store/cifar10.py (hunk 2 above)
Comment (aarati-K) [reply store/cifar10.py:27]: I will make these changes in a follow-up PR.

File: store/cifar10.py (hunk 5 above)
Comment (aarati-K) [reply store/cifar10.py:90]: Will make these changes in a follow-up PR.

File: store/cifar10.py [hunk 6]
```diff
@@ -1,39 +1,125 @@
...
-from sample_creator import SampleCreator
+from store.memory_hierarchy import StorageAttributes, StorageComponents
+from sampling.sample_creator import SampleCreator
 
 
 class Cifar10(DataStore):
     def __init__(self, **kwargs):
         super(Cifar10, self).__init__(**kwargs)
         self.dataset_name = "Cifar-10"
+        self.metadata = Metadata(self).load()
```
Comment (aarati-K) [inline store/cifar10.py:14]: Move this also to base class.

**Pull request merged**